cmd/opm/serve/serve.go

```diff
@@ -88,10 +108,10 @@ func (s *serve) run(ctx context.Context) error {
 		return fmt.Errorf("could not build index model from declarative config: %v", err)
 	}
 	store, err := registry.NewQuerier(m)
-	defer store.Close()
 	if err != nil {
 		return err
 	}
+	defer store.Close()
```

Comment:
In this case, I think we should put this back to _beforE_ the error check.

Under the hood, `registry.NewQuerier` creates a temp dir, and begins marshaling all of the FBC over to the `api.Bundle` format to disk. If an error occurs halfway through, we still return a valid `Querier` on which we can call `Close` to ensure that the temp stuff can be cleaned up, even when where was an error.

If we wait until _after_ the error check, we will no longer clean up the temp dir if an error occurs.

Reply:
Good discussion.  I felt it was probably the case, but generally vscode makes good recommendations so I took this one.  
Reverted in latest.